python/pyspark/context.py

```diff
@@ -128,31 +152,40 @@ class SparkContext(object):
     ValueError: ...
     """
 
-    _gateway = None
-    _jvm = None
+    # set assignment ignore temporarily to prevent errors from other files
+    _gateway: ClassVar[JavaGateway] = None  # type: ignore[assignment]
+    _jvm: ClassVar[JVMView] = None  # type: ignore[assignment]
     _next_accum_id = 0
-    _active_spark_context = None
+    _active_spark_context: ClassVar["SparkContext"] = None  # type: ignore[assignment]
```

Comment:
I am not very fond of this, to be honest.

While we're still discussing possible strategies for handling such cases, for now we should stick to the convention already set by `sql.session`

https://github.com/apache/spark/blob/feba5ac32f2598f6ca8a274850934106be0db64d/python/pyspark/sql/session.py#L289-L290

and use actual types (`Optional[_]`).

However, I've checked this locally and there are almost 500 invocations that require type adjustments if go this way, so I lean towards keeping these ignores for now and creating a separate ticket for further type refinements.

cc @HyukjinKwon @ueshin

Reply:
Yeah .. I agree with keeping ignored..

Reply:
Yeah, let's keep it as is for now and do the refinements in a separate PR.

Reply:
Thanks all!  Unless you have any requests here I'll try to make another pass in the next few days, and if nothing stands out I'll merge it and start working on refinements.